app/controllers/concerns/shopify_app/shop_access_scopes_verification.rb

```diff
@@ -13,11 +13,8 @@ module ShopAccessScopesVerification
 
     def login_on_scope_changes
       if scopes_mismatch?
-        if embedded_param?
```

Comment:
The existing logic here may be wrong. `embedded_param?` is defined as:

```
def embedded_param?
  embedded_redirect_url? && params[:embedded].present? && params[:embedded] == "1"
end
```

and `embedded_redirect_url?` is checking for the presence of the `embedded_redirect_url` configuration value.

But the upgrade guide says this value is optional:
https://github.com/Shopify/shopify_app/blob/main/docs/Upgrading.md#upgrading-to-v2010

Reply:
I think it might still happen on embedded requests, so we might want to handle it. AFAIK `embedded_redirect_url` is optional, but it has a default value so it should still work.

There should be a route in the app that handles the `/exitiframe` path that it defaults to, if this is the setting I think it is.